src/DependencyInjection/ApplicationCompilerPass.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * This file is authored by PrestaShop SA and Contributors <contact@prestashop.com>
+ *
+ * It is distributed under MIT license, since 2021.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Help\PrestaShop\DependencyInjection;
+
+use Help\PrestaShop\RequestInfo;
+use RuntimeException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\Yaml\Yaml;
+
+class ApplicationCompilerPass implements CompilerPassInterface
+{
+    private const FALLBACK_VERSION = '1.7';
+    private const FALLBACK_LANGUAGE = 'en';
+
+    private string $configDir;
+
+    public function __construct(string $configDir)
+    {
+        $this->configDir = rtrim($configDir, '/');
+    }
+
+    public function process(ContainerBuilder $container): void
+    {
+        $requestInfo = RequestInfo::fromRequestUri($_SERVER['REQUEST_URI']);
+
+        $container->set(RequestInfo::class, $requestInfo);
+        $container->setDefinition(RequestInfo::class, (new Definition(RequestInfo::class))->setSynthetic(true));
+
+        $langRepositories = Yaml::parse(file_get_contents($this->getRepositoryMappingFilename()) ?: '');
```

Comment:
This is an expensive operation that could be put in cache. But if we add a cache in front of the webserver it is not needed.